paddle/fluid/framework/details/memory_optimize_helper.cc

```diff
@@ -190,7 +190,7 @@ struct NodeComparator {
     auto rhs_shape = rhs_desc->GetShape();
     if ((lhs_shape[0] == -1 && rhs_shape[0] == -1) ||
         (lhs_shape[0] != -1 && rhs_shape[0] != -1)) {
-      return NodeSize(lhs) <= NodeSize(rhs);
+      return NodeSize(lhs) == NodeSize(rhs);
```

Comment:
what's the memory usage effect of changing this

Reply:
this kind of logic may cause large fragments in the runtime.
Actually in my opinion, current memory_optimize_pass is useless when GC is opened. I can talk about this with you later